File: python/armstrong-numbers/armstrong_numbers.py
def is_armstrong_number(number: int):
    digits: list[int] = []

    remainder = number
    while remainder > 0:
        digits.append(remainder % 10)
        remainder //= 10

    n_digits = len(digits)

    return sum([digit ** n_digits for digit in digits]) == number

File: lua/armstrong-numbers/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
  local digits = {}
  local n = number
  while n > 0 do
    table.insert(digits, 1, n % 10)
    n = math.floor(n / 10)
  end

  for _, digit in ipairs(digits) do
    n = math.floor(n + digit ^ #digits)
  end

  return n == number
end

return ArmstrongNumbers
